Could you please add a test case based on [TF-361](https://bugs.swift.org/browse/TF-361)?

Here's what the reproducer looks like in Python:
```console
$ python
Python 2.7.10 (default, Aug 17 2018, 19:45:58)
[GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.0.42)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> a = object()
>>> a
<object object at 0x1059bf090>
>>> a.hmm = a
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'object' object has no attribute 'hmm'
```

However, currently in Swift, it seems the invalid member access doesn't produce a crash.
```swift
import Python
let x = Python.object()
x.hmm = x // doesn't crash
// print(x) // only crashes when calling another Python function
```

It seems to me the right fix would be to "catch the invalid member access right after it happens" instead of "checking for error state in `dynamicallyCall`". Does this make sense to you?

--- a/stdlib/public/Python/PythonLibrary+Symbols.swift
+++ b/stdlib/public/Python/PythonLibrary+Symbols.swift
@@ -98,7 +98,7 @@ let PyObject_GetAttrString: @convention(c) (
   PythonLibrary.loadSymbol(name: "PyObject_GetAttrString")
 
 let PyObject_SetAttrString: @convention(c) (
-  PyObjectPointer, PyCCharPointer, PyObjectPointer) -> Int =
+  PyObjectPointer, PyCCharPointer, PyObjectPointer) -> Int32 =
   PythonLibrary.loadSymbol(name: "PyObject_SetAttrString")
 
 let PySlice_New: @convention(c) (

--- a/test/Python/python_runtime.swift
+++ b/test/Python/python_runtime.swift
@@ -199,6 +199,11 @@ PythonRuntimeTestSuite.testWithLeakChecking("Errors") {
     // `expectThrows` does not fail if no error is thrown.
     fatalError("No error was thrown.")
   })
+
+  expectCrash(executing: {
+    let a = Python.object()
+    a.foo = "bar"
+  })
 }
 
 PythonRuntimeTestSuite.testWithLeakChecking("Tuple") {

